Packages which depend on "@element-hq/element-web-module-api@workspace:*" cannot be installed

Reproduction case:

```
$ mkdir reproduction-test
$ cd reproduction-test 
$ pnpm add '@element-hq/web-shared-components@0.1.0'
[ERR_PNPM_WORKSPACE_PKG_NOT_FOUND] In ..: "@element-hq/element-web-module-api@workspace:*" is in the dependencies but no package named "@element-hq/element-web-module-api" is present in the workspace

This error happened while installing the dependencies of @element-hq/web-shared-components@0.1.0

Packages found in the workspace: tmp
Progress: resolved 18, reused 14, downloaded 0, added 0

```

The `0.1.0` version of the package does not depend on versioned dependencies but `workspace:*` https://www.npmjs.com/package/@element-hq/web-shared-components?activeTab=code

--- a/.github/workflows/npm-publish.yaml
+++ b/.github/workflows/npm-publish.yaml
@@ -40,4 +40,4 @@ jobs:
 
             - name: 🚀 Publish to npm
               working-directory: packages/${{ inputs.package }}
-              run: npm publish --access public --provenance
+              run: pnpm publish --access public --provenance

